extensions/blocks/logo-gallery/edit.js

```diff
@@ -0,0 +1,306 @@
+					<Toolbar>
+						<MediaUpload
+							onSelect={ this.onSelectImages }
+							allowedTypes={ ALLOWED_MEDIA_TYPES }
+							multiple
+							gallery
+							value={ images.map( img => img.id ) }
+							render={ ( { open } ) => (
+								<IconButton
+									className="components-toolbar__control"
+									label={ __( 'Edit gallery', 'jetpack' ) }
+									icon="edit"
+									onClick={ open }
+								/>
+							) }
+						/>
+					</Toolbar>
+				) }
+			</BlockControls>
+		);
+
+		const toolbarControls = [
+			{
+				icon: 'arrow-down',
+				title: __( 'Small size', 'jetpack' ),
+				isActive: logoSize === 'small',
+				onClick: () => setAttributes( { logoSize: 'small' } ),
+			},
+			{
+				icon: 'sort',
+				title: __( 'Medium size', 'jetpack' ),
+				isActive: logoSize === 'medium',
+				onClick: () => setAttributes( { logoSize: 'medium' } ),
+			},
+			{
+				icon: 'arrow-up',
+				title: __( 'Large size', 'jetpack' ),
+				isActive: logoSize === 'large',
+				onClick: () => setAttributes( { logoSize: 'large' } ),
+			},
+		];
+
+		if ( ! hasImages ) {
+			return (
+				<Fragment>
+					{ controls }
+					<MediaPlaceholder
+						icon={ <BlockIcon icon={ icon } /> }
+						className={ className }
+						labels={ {
+							title: __( 'Logo Gallery', 'jetpack' ),
+							instructions: __(
+								'Drag images, upload new ones or select files from your library.',
+								'jetpack'
+							),
+						} }
+						onSelect={ this.onSelectImages }
+						accept="image/*"
+						allowedTypes={ ALLOWED_MEDIA_TYPES }
+						multiple
+						notices={ noticeUI }
+						onError={ noticeOperations.createErrorNotice }
+					/>
+				</Fragment>
+			);
+		}
+
+		return (
+			<Fragment>
+				{ controls }
+				<InspectorControls>
+					<PanelBody title={ __( 'Logo Gallery Settings', 'jetpack' ) }>
+						<p>{ __( 'Logo size', 'jetpack' ) }</p>
+						<Toolbar controls={ toolbarControls } />
+					</PanelBody>
+				</InspectorControls>
+				{ noticeUI }
+				<ul className={ classNames( className, `is-${ logoSize }` ) }>
+					{ images.map( ( img, index ) => {
+						/* translators: %1$d is the order number of the image, %2$d is the total number of images. */
+						const ariaLabel = sprintf(
+							__( 'image %1$d of %2$d in gallery', 'jetpack' ),
+							index + 1,
+							images.length
+						);
+
+						return (
+							<li className="logo-gallery-item" key={ img.id || img.url }>
+								<Logo
+									url={ img.url }
+									alt={ img.alt }
+									id={ img.id }
+									isSelected={ isSelected && this.state.selectedImage === index }
+									onRemove={ this.onRemoveImage( index ) }
+									onSelect={ this.onSelectImage( index ) }
+									setAttributes={ attrs => this.setImageAttributes( index, attrs ) }
+									aria-label={ ariaLabel }
+								/>
+							</li>
+						);
+					} ) }
+				</ul>
+				<DropZone onFilesDrop={ this.addFiles } />
+				{ isSelected && (
+					<div className="wp-block-jetpack-slideshow__add-item">
```

Comment:
Note class name of another block. ;-)

Since this upload thing is used in Gallery, slideshow and here, you might want to consider turning into a shared component between the three. :-)

Reply:
Good catch! I'll quick fix this first and add the shared component to the TODOs.

Reply:
See also how core changed their upload button for gallery block: https://github.com/WordPress/gutenberg/pull/12367

Reply:
Nice, I hadn't seen that one. I've noted it in the TODOs! 👍